### PR TITLE
Added DCO sign-off hook and updated info regarding the migration from CLA to DCO

### PR DESCRIPTION
**Pull request by hbelmiro**
The DCO sign-off hook was copied from [Argo Project](https://github.com/argoproj/argoproj/blob/main/community/CONTRIBUTING.md#legal).

### CODE REVIEW

File: dco-signoff-hook/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Signing off commits
+
+To automatically sign off on every commit, copy the [prepare-commit-msg](prepare-commit-msg) file to the `.git/hooks` directory in your repo or if you already have such a hook, merge the contents into your existing hook.
```
Comment (terrytangyuan) [inline dco-signoff-hook/README.md:3]: Can you also add instructions on how to do this manually? e.g. `git commit -s -m 'This is my commit message'`

**Review by terrytangyuan — COMMENTED** [review]
Thank you!

File: dco-signoff-hook/README.md (hunk 1 above)
Comment (terrytangyuan) [reply dco-signoff-hook/README.md:3]: Similar to https://github.com/kubeflow/website/pull/3682

Comment (hbelmiro) [reply dco-signoff-hook/README.md]: Done.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by diegolovison** [comment]
I was going to say to close and not merge because of https://github.com/kubeflow/community/pull/703

**Pull request merged**